MaxText/pyconfig.py

```diff
@@ -129,6 +139,33 @@ def user_init(raw_keys):
 
     validate_attention_type(raw_keys['attention'])
 
+  @staticmethod
+  def update_model_vars(raw_keys):
+    ''' Update model config variables
+    '''
+    validate_model_name(raw_keys['model_name'])
+    if raw_keys['model_name'] == 'llama-7b':
+      max_logging.log(f"Running Model: {raw_keys['model_name']}")
+      llama_7b_model_vars = {
+        'base_emb_dim': 4096,
+        'base_num_heads': 32,
+        'base_mlp_dim': 11008,
+        'base_num_decoder_layers': 32,
+        'head_dim': 128,
+        'mlp_activations': ['silu'],
+      }
+      raw_keys = validate_and_update_keys(raw_keys, llama_7b_model_vars)
+
+def validate_and_update_keys(raw_keys, model_keys):
+  max_logging.log("Updating following parameters in config\n")
+  for k in model_keys:
+    max_logging.log(f"{k}: {model_keys[k]}")
+    if k in raw_keys and isinstance(raw_keys[k], type(model_keys[k])):
+      raw_keys[k] = model_keys[k]
+    else:
+      raise ValueError(f'Key {k} does not exist in config/base.yml.')
```

Comment:
Give a more polite error message! I.e. it might be the wrong type. A standard way to write this beautifully is:

```
if k not in raw_keys:
  raise ValueError("Blah1")
elif  not isinstance
  raise ValueError("Blah2")
else:
  do the thing